src/map/map.ts

```diff
@@ -876,7 +913,11 @@ export class PropertiesMap {
      */
     private _xValues(trace?: number): number[][] {
         const values = this._properties()[this._current.x].values;
-        const selected = [values[this._selected]];
+        const selected = [];
+        for (const guid of this._guids) {
+          selected.push(values[this._markerMap[guid].current]);
+        }
+
         if (!this._is3D()) {
```

Comment:
Just realised this needs to be adapted with the code above, setting all selected markers positions to NaN in 2D mode (and not only the first one), since in 2D mode we are manually dealing with the markers instead of leaving it to plotly for performance reasons.

Reply:
The initial bug this was fixing occurred when going from 2D to 3D and then back to 2D, markers from the 3D mode where not correctly removed

Reply:
Can you elaborate? Not sure of the changes needed.